Source/WebCore/platform/mediastream/WebRTCProvider.h

```diff
@@ -0,0 +1,137 @@
+ *    notice, this list of conditions and the following disclaimer in the
+ *    documentation and/or other materials provided with the distribution.
+ *
+ * THIS SOFTWARE IS PROVIDED BY APPLE INC. AND ITS CONTRIBUTORS ``AS IS''
+ * AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO,
+ * THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR
+ * PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL APPLE INC. OR ITS CONTRIBUTORS
+ * BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR
+ * CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF
+ * SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS
+ * INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN
+ * CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE)
+ * ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF
+ * THE POSSIBILITY OF SUCH DAMAGE.
+ */
+
+#pragma once
+
+#include "MDNSRegisterError.h"
+#include "RTCDataChannelRemoteHandlerConnection.h"
+#include "RTCRtpCapabilities.h"
+#include "ScriptExecutionContextIdentifier.h"
+#include <wtf/CompletionHandler.h>
+#include <wtf/Expected.h>
+#include <wtf/UniqueRef.h>
+#include <wtf/text/WTFString.h>
+
+namespace WebCore {
+
+class ContentType;
+struct MediaCapabilitiesInfo;
+struct VideoConfiguration;
+struct MediaCapabilitiesDecodingInfo;
+struct MediaCapabilitiesEncodingInfo;
+struct MediaDecodingConfiguration;
+struct MediaEncodingConfiguration;
+
+class WebRTCProvider {
+    WTF_MAKE_FAST_ALLOCATED;
+public:
+    WEBCORE_EXPORT static UniqueRef<WebRTCProvider> create();
+    WebRTCProvider() = default;
+    virtual ~WebRTCProvider() = default;
+
+    WEBCORE_EXPORT static bool webRTCAvailable();
+    WEBCORE_EXPORT static void setH264HardwareEncoderAllowed(bool);
+
+    virtual void setActive(bool);
+
+    using IPAddressOrError = Expected<String, MDNSRegisterError>;
+    using MDNSNameOrError = Expected<String, MDNSRegisterError>;
+
+    virtual RefPtr<RTCDataChannelRemoteHandlerConnection> createRTCDataChannelRemoteHandlerConnection() { return nullptr; }
+
+    using DecodingConfigurationCallback = Function<void(MediaCapabilitiesDecodingInfo&&)>;
+    using EncodingConfigurationCallback = Function<void(MediaCapabilitiesEncodingInfo&&)>;
+    void createDecodingConfiguration(MediaDecodingConfiguration&&, DecodingConfigurationCallback&&);
+    void createEncodingConfiguration(MediaEncodingConfiguration&&, EncodingConfigurationCallback&&);
+
+    WEBCORE_EXPORT void setH265Support(bool);
+    WEBCORE_EXPORT void setVP9Support(bool supportsVP9Profile0, bool supportsVP9Profile2);
+    WEBCORE_EXPORT void setVP9VTBSupport(bool);
+    bool isSupportingH265() const { return m_supportsH265; }
+    bool isSupportingVP9Profile0() const { return m_supportsVP9Profile0; }
+    bool isSupportingVP9Profile2() const { return m_supportsVP9Profile2; }
+    virtual bool isSupportingVP9VTB() const { return m_supportsVP9VTB; }
```

Comment:
VTB is COCOA specific. Maybe move it back to LibWebRTC as well?

Reply:
Right, I thought about it indeed :) the 3 others isSupporting methods are fine in the base class though, imho.